### PR TITLE
Drop outdated documentation

### PR DESCRIPTION
**Pull request by pirj**
Mostly related to either RSpec Rails 3 or even earlier.

### CODE REVIEW

File: features/GettingStarted.md [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+    $ cd example_app
 
 ### Add rspec-rails to the Gemfile
 
-    $ echo 'gem "rspec-rails", :group => [:development, :test]' >> Gemfile
+    $ echo 'gem "rspec-rails", group: [:development, :test]' >> Gemfile
 
 ### Install the bundle
 
-    $ bundle install
```
Comment (pirj) [inline features/GettingStarted.md:16]: `install` is not enough as it doesn't add to `Gemfile.lock`.

File: features/GettingStarted.md [hunk 2]
```diff
@@ -29,15 +29,15 @@ the files in the `spec` directory to RSpec.
 
 ### Run migrations
 
-    $ rake db:migrate && rake db:test:prepare
+    $ rails db:migrate && rails db:test:prepare
 
 ### Run RSpec
 
     $ rake spec
```
Comment (pirj) [inline features/GettingStarted.md:36]: Never using this, not sure if it can/should be `rails spec` now or `rake` is preferrable just like with running other common tasks.

File: features/GettingStarted.md [hunk 3]
```diff
@@ -29,15 +29,15 @@ the files in the `spec` directory to RSpec.
...
 
 or
 
-    $ rspec spec --format documentation
```
Comment (pirj) [inline features/GettingStarted.md:40]: Doc format is the default when running just one file.
`spec` is the default directory.

File: features/RailsVersions.md [hunk 4]
```diff
@@ -8,3 +9,5 @@
     4.2           | >= 3.1
     5.0           | >= 3.5
     5.1           | >= 3.6
+    5.2           | >= 3.8
```
Comment (pirj) [inline features/RailsVersions.md:12]: It might be that Rails 5.2 was supported in 3.6/3.7, but in 3.8 there were some fixes to improve Rails 5.2 support.

File: features/upgrade/README.md [hunk 5]
```diff
@@ -1,178 +1,5 @@
-# Upgrading from rspec-rails-2.x to rspec-rails-3
+# Upgrading from rspec-rails 3.x to version 4
 
-For detailed information on the general RSpec 3.x upgrade process see the [RSpec
-Upgrade docs](https://relishapp.com/rspec/docs/upgrade).
+There is nothing that needs to be done to upgrade from RSpec Rails 3 to version 4.0.
```
Comment (pirj) [inline features/upgrade/README.md:3]: Can't think of anything, really.
Should we mention that the minimum Ruby version is now 2.2, and we recomment to update Rails 4.2 to a newer version?

File: features/.nav [hunk 6]
```diff
@@ -1,5 +1,3 @@
-- upgrade:
-  - from_1x_to_2x.md (From rspec-rails-1.x to rspec-rails-2)
```
Comment (pirj) [inline features/.nav:2]: Ooops, leftover.

File: features/upgrade/README.md [hunk 7]
```diff
@@ -1,178 +1,5 @@
...
-`ActiveRecord` behavior complexity which would otherwise be good to expose.
-
-Some alternatives are:
-
-- Wrap calls to `ActiveRecord` objects in more specific domain models and
-  services
-- Use new unsaved `ActiveRecord` instances (e.g. `Model.new`)
-- Consider [partial mocks](https://www.relishapp.com/rspec/rspec-mocks/docs/basics/partial-test-doubles) on an `ActiveRecord` instance
-- Let the specs hit database directly where appropriate
+If you are upgrading from an earlier version of RSpec Rails, please consult [this document](https://relishapp.com/rspec/rspec-rails/v/3-9/docs/upgrade).
```
Comment (JonRowe) [inline features/upgrade/README.md:5]:
```suggestion
If you are upgrading from an earlier version of RSpec Rails, please consult [the upgrading 2.x to 3.x guide](https://relishapp.com/rspec/rspec-rails/v/3-9/docs/upgrade).
```

File: features/directory_structure.feature [hunk 8]
```diff
@@ -63,7 +63,7 @@ Feature: Directory Structure
   their location on the filesystem. This was both confusing to new users and not
   desirable for some veteran users.
 
-  In RSpec 3, this behavior must be explicitly enabled:
+  In RSpec 4, this behavior must be explicitly enabled:
```
Comment (JonRowe) [inline features/directory_structure.feature:66]:
```suggestion
  This behaviour must be explicitly enabled:
```

File: features/RailsVersions.md (hunk 4 above)
Comment (JonRowe) [reply features/RailsVersions.md:12]: 👍

File: features/GettingStarted.md (hunk 3 above)
Comment (JonRowe) [reply features/GettingStarted.md:40]: Explicit is better :/

File: features/GettingStarted.md (hunk 2 above)
Comment (JonRowe) [reply features/GettingStarted.md:36]: Is rails available in 5?

Comment (pirj) [reply features/GettingStarted.md]: It is https://github.com/rails/rails/blob/5-0-stable/guides/source/command_line.md#binrails

File: features/upgrade/README.md (hunk 5 above)
Comment (benoittgt) [reply features/upgrade/README.md:3]: I think it's a good idea :)

File: features/GettingStarted.md [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
 
-    $ rails new example
-    $ cd example
+    $ rails new example_app
+    $ cd example_app
 
 ### Add rspec-rails to the Gemfile
 
-    $ echo 'gem "rspec-rails", :group => [:development, :test]' >> Gemfile
+    $ echo 'gem "rspec-rails", group: [:development, :test]' >> Gemfile
```
Comment (benoittgt) [inline features/GettingStarted.md:12]: Side note but isn't against properly organizing your Gemfile?

Comment (pirj) [reply features/GettingStarted.md:12]: ¯\\\_(ツ)_/¯

File: features/GettingStarted.md (hunk 1 above)
Comment (JonRowe) [reply features/GettingStarted.md:16]: Can you clarify? I thought `bundle` was short for `bundle install` and should be sufficient for a new app...

File: features/upgrade/README.md (hunk 5 above)
Comment (JonRowe) [reply features/upgrade/README.md:3]: Yes this should mention that Rspec Rails is intended for Rails 5 and 6 and that the minimum ruby version is 2.2

File: features/GettingStarted.md (hunk 9 above)
Comment (JonRowe) [reply features/GettingStarted.md:12]: One liner is a one liner :joy:

File: features/upgrade/README.md [hunk 10]
```diff
@@ -1,178 +1,9 @@
...
 
-For detailed information on the general RSpec 3.x upgrade process see the [RSpec
-Upgrade docs](https://relishapp.com/rspec/docs/upgrade).
+RSpec Rails 4 requires Ruby 2.2.
 
-There are several changes specific to `rspec-rails` to be aware of:
+Please also keep in mind that even though RSpec Rails 4 has soft support for Rails 4.2, it's recommended to use Rails 5+.
 
-- [Default helper files created in RSpec 3.x have changed](#default-helper-files)
+Other than that, there is nothing that needs to be done to upgrade from RSpec Rails 3 to version 4.0.
```
Comment (JonRowe) [inline features/upgrade/README.md:7]: I would rephrase this, I feel its important to emphasise the last part, and less so the first part.

```suggestion
RSpec Rails 4 is a major version under semantic versioning, it allowed us to change the supported Rails versions to 5 and 6 only. There are no changes required to upgrade to RSpec Rails 4 if you are using Rails 5 or 6.

If you are using Rails 4.2 you can use RSpec Rails 4, but note that support for it is not maintained, we consider this a breaking change hence the version change, and you must be on Ruby 2.2 as a minimum.
```

Comment (pirj) [reply features/upgrade/README.md:7]: Much better, thanks.

File: features/GettingStarted.md (hunk 1 above)
Comment (pirj) [reply features/GettingStarted.md:16]: > Install the gems specified in your Gemfile(5). If this is the first time you run bundle install (and a Gemfile.lock does not exist), Bundler will fetch all remote sources, resolve dependencies and install all needed gems.

You're right. I had an impression that `bundle install` is not supposed to create/update `Gemfile.lock`. It seems that it's only the case for `bundle install --frozen`.
Will revert this.
